app/models/course/condition/achievement.rb

```diff
@@ -3,4 +3,27 @@ class Course::Condition::Achievement < ActiveRecord::Base
   belongs_to :achievement, class_name: Course::Achievement.name
 
   default_scope { includes(:achievement) }
+
+  validates :achievement, presence: true
+  validate :valid_achievement_condition
+
+  def name
+    achievement.title
+  end
+
+  private
+
+  def valid_achievement_condition
```

Comment:
Assignment Branch Condition size for valid_achievement_condition is too high. [15.56/15]<br>Method has too many lines. [11/10]